controllers/nicclusterpolicy_controller.go

```diff
@@ -249,20 +253,39 @@ func (r *NicClusterPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}
 	r.stateManager = stateManager
 
-	builder := ctrl.NewControllerManagedBy(mgr).
+	ctl := ctrl.NewControllerManagedBy(mgr).
 		For(&mellanoxv1alpha1.NicClusterPolicy{}).
+		WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
```

Comment:
is it related to the issue at hand ?
what is the default if not provided ?

in any case, im ok with limiting reconciles to 1

Reply:
default is one, I think it is safe to drop this option

Reply:
> default is one, I think it is safe to drop this option

thats what i remembered :)